lucet-runtime/lucet-runtime-internals/src/module.rs

```diff
@@ -143,32 +143,36 @@ pub trait ModuleInternal: Send + Sync {
     /// Returns a `Result<(), Error>` rather than a boolean in order to provide a richer accounting
     /// of what may be invalid.
     fn validate_runtime_spec(&self, limits: &Limits) -> Result<(), Error> {
-        let heap = self.heap_spec();
-        // Assure that the total reserved + guard regions fit in the address space.
-        // First check makes sure they fit our 32-bit model, and ensures the second
-        // check doesn't overflow.
-        if heap.reserved_size > std::u32::MAX as u64 + 1
-            || heap.guard_size > std::u32::MAX as u64 + 1
-        {
-            return Err(lucet_incorrect_module!(
-                "heap spec sizes would overflow: {:?}",
-                heap
-            ));
-        }
+        // Modules without heap specs will not access the heap
```

Comment:
WebAssembly module validation enforces this property

Reply:
Gotcha. I guess we'll have to write some `.wat` tests manually in order to test this? I can't imagine clang could be convinced to produce a memory-free module.

Reply:
We can't even write the tests in wat, because the `wabt` library will reject the wat programs while translating them to wasm. The spec tests do provide coverage of these cases, though: https://github.com/WebAssembly/spec/blob/master/test/core/memory.wast#L22